### PR TITLE
Migrate Google Maps to OpenStreetMap

### PR DESCRIPTION
**Pull request by McGiverGim**
Fixes https://github.com/betaflight/betaflight-configurator/issues/1303

It seems a little slower than the Google version, but it works, and is open ;)

It uses OpenLayers to manage OpenStreetMap. Some of the code has been taken from iNav (thanks!).

Some things:
- I removed the manifest permission for Google Maps api, but I didn't need to add the Open Street Map permission. Strange.
- I tried to add the OpenLayers package using npm, but the version downloaded was in a modules form and I have not been able to make it work as a Chrome App, so I added manually the standard 5.3.0 javascript library. Maybe other can make it work in the future.
- I have changed completely the map.js file, sorry for that, so it is better to ignore the changes shown in github and take a look at directly to the file, it is not too long and will be easier to review.

### CODE REVIEW

**Comment by mikeller** [comment]
This looks good. I'll do some tests and a more thorough review of the code later on.

Seeing that there is no web version of the OpenLayers framework available on npm I think adding the minified version into the repository is the best approach for now.

We've had the same problem (i.e. making it work for the Chrome Web Store) with other libraries before. We potentially should invest some time after the next release to set up something like browserify to allow us to bundle arbitrary npm packages into a version that works in the Chrome Web app as part of the build.
Also, I've added analytics collection that will allow us to accurately gauge how many users are still using the Chrome Web app from the next configurator release onwards, which will make it easier for us to decide on how much effort to spend on this.

File: manifest.json [hunk 1]
```diff
@@ -18,7 +18,6 @@
         }
     },
     "permissions": [
-        "https://maps.googleapis.com/*",
```
Comment (mikeller) [inline manifest.json]: The reason you did not seem to have to add anything here is that NW.js does not use `manifest.json`. The Chrome Web App does, and for it to load the map you'll have to add `https://api.openstreetmap.org/` in here.

The good news is that once this is done, the map loads and displays just fine in the Chrome Web App. Good work!

File: libraries/openlayers/ol.css.map [hunk 2]
```diff
@@ -0,0 +1 @@
...AAG,OAAO,KAExD,gBACE,IAAK,QACL,KAAM,KAER,gBACE,MAAO,KACP,IAAK,KAEP,aACE,YACE,QAAS,MAIb,mBACE,QAAS,MACT,OAAQ,IACR,QAAS,EACT,MAAO,KACP,UAAW,OACX,YAAa,IACb,gBAAiB,KACjB,WAAY,OACZ,OAAQ,QACR,MAAO,QACP,YAAa,KACb,iBAAkB,kBAClB,OAAQ,KACR,cAAe,IAEjB,qCACE,OAAQ,KACR,QAAS,EAEX,uBACE,YAAa,MAEf,YACE,QAAS,MACT,YAAa,IACb,UAAW,MACX,YAAa,UAEf,6BACE,UAAW,MAEb,0BACE,IAAK,MAGP,yBADA,yBAEE,gBAAiB,KACjB,iBAAkB,kBAEpB,qBACE,cAAe,IAAI,IAAI,EAAE,EAE3B,sBACE,cAAe,EAAE,EAAE,IAAI,IAIzB,gBACE,WAAY,MACZ,OAAQ,KACR,MAAO,KACP,UAAW,mBAGb,mBACE,OAAQ,EACR,QAAS,EAAE,KACX,UAAW,MACX,YAAa,QACb,MAAO,KACP,YAAa,EAAE,EAAE,IAAI,KAEvB,mBACE,QAAS,OACT,WAAY,KACZ,YAAa,QAEf,0CACE,QAAS,IAEX,oBACE,WAAY,IACZ,UAAW,QACX,eAAgB,OAEE,uBAApB,mBACE,QAAS,aAEX,gCACE,QAAS,KAEX,mCACE,WAAY,qBAEd,iCACE,OAAQ,EACR,MAAO,EACP,cAAe,IAAI,EAAE,EACrB,OAAQ,MACR,YAAa,IAEf,qCACE,WAAY,MACZ,WAAY,MAEd,wCACE,QAAS,KAGX,eACE,IAAK,MACL,KAAM,KACN,OAAQ,MAEV,sBACE,SAAU,SACV,OAAQ,KAGV,yBACE,IAAK,MAGP,gBACE,KAAM,KACN,OAAQ,KAEV,iCACE,OAAQ,EACR,KAAM,EACN,cAAe,EAAE,IAAI,EAAE,EAEzB,oCACA,uBACE,QAAS,aAEX,oCACE,OAAQ,IAAI,MAAM,QAClB,OAAQ,MACR,OAAQ,IACR,MAAO,MAET,0CACE,OAAQ,IACR,KAAM,IACN,SAAU,SAEZ,iDACA,wCACE,QAAS,KAEX,mCACE,WAAY,qBAEd,oBACE,OAAQ,IAAI,OAAO,kBAGrB,0CACE,OAAQ"}
```
Comment (mikeller) [inline libraries/openlayers/ol.css.map]: Not sure if we actually need the map files - have you tried running without them?

File: src/js/tabs/map.js [hunk 3]
```diff
@@ -1,76 +1,86 @@
...
-         map.setCenter(new google.maps.LatLng(e.data.lat, e.data.lon));
-         marker.setPosition( new google.maps.LatLng( e.data.lat, e.data.lon ) );
-         map.panTo( new google.maps.LatLng( e.data.lat, e.data.lon ) );
-    }
-  } catch (e) {
-    console.log('message error');
-  }
-});
+const DEFAULT_ZOOM = 16,
+      DEFAULT_LON = 10.001362,
```
Comment (mikeller) [inline src/js/tabs/map.js]: Not sure you want to tell everybody where you live. ;-)

File: libraries/openlayers/ol.css.map (hunk 2 above)
Comment (McGiverGim) [reply libraries/openlayers/ol.css.map]: No, I simply added the complete library. I can remove and test it...

File: manifest.json (hunk 1 above)
Comment (McGiverGim) [reply manifest.json]: I tested the chrome app but is clear that I did something wrong... I will add it.

File: src/js/tabs/map.js (hunk 3 above)
Comment (McGiverGim) [reply src/js/tabs/map.js]: They are not mine, there were in the old version too... maybe is better to let them as zero, like iNav does...

File: manifest.json (hunk 1 above)
Comment (McGiverGim) [reply manifest.json]: Added...

File: libraries/openlayers/ol.css.map (hunk 2 above)
Comment (McGiverGim) [reply libraries/openlayers/ol.css.map]: Removed, all seems to work.

File: src/js/tabs/map.js (hunk 3 above)
Comment (McGiverGim) [reply src/js/tabs/map.js]: Changed to lon lat zero, zero

**Comment by McGiverGim** [comment]
Fixed as per review

**Comment by mikeller** [comment]
Looking all good now, thanks!

**Pull request merged**